istioctl/pkg/install/k8sversion/version.go

```diff
@@ -38,7 +39,7 @@ func CheckKubernetesVersion(versionInfo *version.Info) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	return MinK8SVersion <= v, nil
+	return (MinK8SVersion <= v && MaxK8SVersion >= v), nil
```

Comment:
Not sure we want to add maxK8sVersion check or not. @therealmitchconnors @esnible  any thoughts for this?

Reply:
Problem might be that TOC has stated that we should try supporting the next k8s release after release if it's a short time. Example is 1.12 should be expanded to support 1.23 (discussion was for the RMs to have a long running qualification test done on the new k8s version when it comes out).  Certainly, the main branch should support 1.23, and would need to likely support 1.24 at it's release (likely on the istio 1.13 branch).